### PR TITLE
Modifies wxcode check_tree utility function to report issues with unreachable nodes

### PR DESCRIPTION
**Pull request by MoseleyS**
Modifies wxcode check_tree utility function to report issues with unreachable nodes

- Unit-test to show this works
- Acceptance test to show this works at CLI level
- Master CLI test wrapper modified to return whatever the CLI returns so these strings can be tested.

Addresses https://github.com/metoppv/mo-blue-team/issues/161

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver_tests/wxcode/wxcode/test_utilities.py [hunk 1]
```diff
@@ -518,6 +519,15 @@ def test_check_tree(modify_tree, expected):
...
 
+def test_check_tree_unreachable_leaf():
+    """Add an unreachable leaf and check for an error."""
+    expected = "Unreachable node 'kittens'"
+    tree = wxcode_decision_tree()
+    tree["kittens"] = tree["lightning"].copy()
+    result = check_tree(tree)
+    assert result == expected
+
+
```
Comment (bayliffe) [inline improver_tests/wxcode/wxcode/test_utilities.py:530]: This test could be replaced with an extension of the above parameterisation, e.g.
```
        (
            "mist_conditions",
            "if_true",
            "no_precipitation_cloud",
            "Unreachable node 'fog_conditions'",
        ),
```

File: improver_tests/acceptance/acceptance.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def run_cli(cli_name, verbose=True):
     def run_function(args):
         if not checksum_ignore():
             verify_checksums(args)
-        cli.main("improver", cli_name, *args, verbose=verbose)
+        return cli.main("improver", cli_name, *args, verbose=verbose)
```
Comment (bayliffe) [inline improver_tests/acceptance/acceptance.py:72]: For next reviewer, this is needed to return the text output from the `--check-tree` option as this does not raise an exception. An exception could be raised as an alternative, if the tree is invalid, and then the test could be for an exception and this return would not be needed, but I think I prefer it as is.

**Review by bayliffe — CHANGES_REQUESTED** [review]
Just one unit test change request to simplify things.

File: improver_tests/wxcode/wxcode/test_utilities.py (hunk 1 above)
Comment (MoseleyS) [reply improver_tests/wxcode/wxcode/test_utilities.py]: Good idea. I didn't think of this. Now done.

**Review by bayliffe — DISMISSED** [review]

File: improver_tests/acceptance/test_wxcode.py [hunk 3]
```diff
@@ -177,3 +178,23 @@ def test_no_lightning(tmp_path):
...
+
+@pytest.mark.parametrize(
+    "wxtree,expected",
+    (
+        ("wx_decision_tree_1h.json", "Decision tree OK\nRequired inputs are:"),
+        ("wx_decision_tree_3h.json", "Decision tree OK\nRequired inputs are:"),
+        ("bad_wx_decision_tree.json", "Unreachable node 'unreachable'"),
+    ),
+)
+def test_trees(wxtree, expected):
```
Comment (fionaRust) [inline improver_tests/acceptance/test_wxcode.py:191]: I think this should be a unit test not an acceptance tests. The rest of the utility you've modified has been unit tested not acceptance tested.

**Review by fionaRust — CHANGES_REQUESTED** [review]

File: improver_tests/acceptance/test_wxcode.py (hunk 3 above)
Comment (MoseleyS) [reply improver_tests/acceptance/test_wxcode.py:200]: The point of this test is to check that the decision tree used in the other acceptance tests is valid, as well as making sure that the option works when called from the CLI.
I've just updated this branch with the latest changes from master, as Ben's change to unify the decision trees would have broken this after merging.

Comment (fionaRust) [reply improver_tests/acceptance/test_wxcode.py:200]: Ok, that makes sense for the valid decision tree, but I think the failure should still be tested in the same place as all the other failures this bit of code can detect, which is in the unit tests.

File: improver_tests/wxcode/wxcode/test_utilities.py [hunk 4]
```diff
@@ -590,6 +591,12 @@ def modify_tree_fixture(node, key, value):
             ["kittens"],
             "Node lightning has a non-numeric probability threshold ['kittens']",
         ),
+        (
+            "mist_conditions",
+            "if_true",
+            "no_precipitation_cloud",
+            "Unreachable node 'fog_conditions'",
+        ),
```
Comment (MoseleyS) [inline improver_tests/wxcode/wxcode/test_utilities.py:599]: @fionaRust: The unreachable node check is tested in this unit test, as well as the acceptance test. Do you think the acceptance test is unnecessary?

Comment (fionaRust) [reply improver_tests/wxcode/wxcode/test_utilities.py:599]: I guess so, I'm not sure we should test it in both places, and I keep dipping in and out of looking at this PR and getting confused about what's tested where and why. I'm not too bothered though, so I'm going to approve it now anyway and let you decide if you want to remove the failure acceptance test or not.

**Review by fionaRust — APPROVED** [review]

**Review by fionaRust — APPROVED** [review]

**Comment by MoseleyS** [comment]
Acceptance test data also merged.

**Pull request merged**